xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/BlockchainAdapters.java

```diff
@@ -86,4 +108,144 @@ public static CurrencyPair toCurrencyPairBySymbol(BlockchainSymbol blockchainSym
+                    .originalAmount(blockchainOrder.getCumQty().add(blockchainOrder.getLeavesQty()))
+                    .id(Long.toString(blockchainOrder.getExOrdId()))
+                    .timestamp(blockchainOrder.getTimestamp())
+                    .averagePrice(blockchainOrder.getAvgPx())
+                    .build();
+
+            if (order instanceof LimitOrder) {
+                limitOrders.add((LimitOrder) order);
+            } else {
+                hiddenOrders.add(order);
+            }
+        }
+
+        return new OpenOrders(limitOrders, hiddenOrders);
+    }
+
+    public static Order toOpenOrdersById(BlockchainOrder blockchainOrder){
+        Order.Builder builder = blockchainOrder.getOrderBuilder();
+
+        return builder.originalAmount(blockchainOrder.getCumQty().add(blockchainOrder.getLeavesQty()))
+                      .id(Long.toString(blockchainOrder.getExOrdId()))
+                      .timestamp(blockchainOrder.getTimestamp())
+                      .averagePrice(blockchainOrder.getAvgPx())
+                      .cumulativeAmount(blockchainOrder.getCumQty())
+                      .orderStatus(toOrderStatus(blockchainOrder.getOrdStatus()))
+                      .userReference(blockchainOrder.getClOrdId())
+                      .build();
+    }
+
+    public static Order.OrderStatus toOrderStatus(String status) {
+        switch (status.toUpperCase()) {
+            case OPEN:
+                return Order.OrderStatus.OPEN;
+            case REJECTED:
+                return Order.OrderStatus.REJECTED;
+            case CANCELED:
+                return Order.OrderStatus.CANCELED;
+            case FILLED:
+                return Order.OrderStatus.FILLED;
+            case PART_FILLED:
+                return Order.OrderStatus.PARTIALLY_FILLED;
+            case EXPIRED:
+                return Order.OrderStatus.EXPIRED;
+            case PENDING:
+                return Order.OrderStatus.PENDING_NEW;
+            default:
+                return Order.OrderStatus.UNKNOWN;
+        }
+    }
+
+    public static BlockchainOrder toBlockchainLimitOrder(LimitOrder limitOrder){
+        return BlockchainOrder.builder()
+                .ordType(LIMIT)
+                .symbol(toCurrencyPair(limitOrder.getInstrument()))
+                .side(Order.OrderType.BID.equals(limitOrder.getType())? BUY.toUpperCase() : SELL.toUpperCase())
+                .orderQty(limitOrder.getOriginalAmount())
+                .price(limitOrder.getLimitPrice())
+                .clOrdId(generateClOrdId())
+                .build();
+    }
+
+    public static BlockchainOrder toBlockchainMarketOrder(MarketOrder marketOrder){
+        return BlockchainOrder.builder()
+                .ordType(MARKET)
+                .symbol(toCurrencyPair(marketOrder.getInstrument()))
+                .side(Order.OrderType.BID.equals(marketOrder.getType())? BUY.toUpperCase() : SELL.toUpperCase())
```

Comment:
You are doing the same thing, here in line 196 and in line 174. 
Move this logic to a method called "getSideFromOrderType" and use it in all cases.